pkg/applicationserver/io/packages/redis/registry.go

```diff
@@ -215,16 +216,18 @@ func (r ApplicationPackagesRegistry) Set(ctx context.Context, ids ttnpb.Applicat
 					return errInvalidIdentifiers
 				}
 			} else {
-				if err := ttnpb.ProhibitFields(sets,
-					"ids.end_device_ids.application_ids",
-					"ids.end_device_ids.device_id",
-					"ids.f_port",
-				); err != nil {
-					return errInvalidFieldmask.WithCause(err)
-				}
 				if err := cmd.ScanProto(updated); err != nil {
 					return err
 				}
+				if ttnpb.HasAnyField(sets, "ids.end_device_ids.application_ids.application_id") && pb.ApplicationIdentifiers.ApplicationID != updated.ApplicationID {
```

Comment:
`updated` -> `stored`

Reply:
- `pb` is the proto being set(input)
- `stored` is the proto currently in the registry
- `updated` is the result of applying `pb` with specified fieldmask to `stored`